aten/src/ATen/native/cuda/SoftMax.cu

```diff
@@ -371,29 +375,40 @@ blockReduce(AccumT* smem, AccumT val,
 
 template <template<typename, typename> class Reduction, int ILP, typename T, typename AccumT>
 __device__ __forceinline__ AccumT
-ilpReduce(T* data,
+ilpReduce(int shift,
+          T* data,
           int size,
           const Reduction<T, AccumT>& r,
           AccumT defaultVal)
 {
+  typedef typename std::aligned_storage<ILP*sizeof(T), ILP*alignof(T)>::type LoadT;
```

Comment:
Why are you using `aligned_storage` at some places while using `aligned_vector` else? IIRC, MSVC has some issue with `aligned_storage` that failed your vectorized dropout PR?

Reply:
Good point - this is code ported from an apex routine by Deyu - I missed that one